### PR TITLE
add support for pmrun by dell

### PR DESCRIPTION
**Pull request by juliedavila**
##### ISSUE TYPE

Feature Pull Request
##### ANSIBLE VERSION

ansible 2.2.0 (pmrun c708c21e61) last updated 2016/06/16 17:49:57 (GMT -700)
  lib/ansible/modules/core: (detached HEAD 1d0f408897) last updated 2016/06/14 10:10:53 (GMT -700)
  lib/ansible/modules/extras: (detached HEAD 146c969c07) last updated 2016/06/14 10:10:53 (GMT -700)
  config file =
  configured module search path = Default w/o overrides
##### SUMMARY

Adds support for no-password pmrun priv escalation

### CODE REVIEW

File: lib/ansible/playbook/play_context.py [hunk 1]
```diff
@@ -539,6 +539,17 @@ def detect_su_prompt(data):
 
                 becomecmd = '%s -u %s %s' % (exe, self.become_user, command)
 
+            elif self.become_method == 'pmrun':
+                #TODO: Add support for prompt
```
Comment (bcoca) [inline lib/ansible/playbook/play_context.py:543]: so 'Password:' is not a valid prompt? you should leave unset in that case

**Comment by bcoca** [comment]
To use this it also requires whitelisting at the connection plugin level.

**Comment by wtcross** [comment]
@bcoca @defionscode Do one of you have a way to test this code? I think that adding password support is a great idea, but we'll need to test that it's working before merging.

**Comment by wtcross** [comment]
I'm happy to help out with this PR, btw.

File: lib/ansible/playbook/play_context.py (hunk 1 above)
Comment (juliedavila) [reply lib/ansible/playbook/play_context.py:543]: it is a valid prompt

Comment (wtcross) [reply lib/ansible/playbook/play_context.py:543]: If this is the valid prompt, then isn't it going to work when pmrun prompts for a password? My understanding of this bit of code isn't very good, but it seems like you just set the `prompt` variable and then password input is handled for you.

@bcoca is this a correct assumption?

**Comment by wtcross** [comment]
The `pmrun` priv escalation method has been whitelisted at the connection plugin level.

**Comment by juliedavila** [comment]
@wtcross looks like your pr to my fork caused some merge conflicts

**Pull request closed without merging**